### PR TITLE
HZN-626: Collect JMS queue statistics from the embedded broker via JMX

### PR DESCRIPTION
**Pull request by pk015603**
HZN - 626: Collect JMS queue statistics from the embedded broker via JMX:
    This JIRA enhanced to capture the JMX data collection facility for the ActiveMQ. Covered the following attrbutes:
1. Connection count.
2. Consumer count.
3. Message count.
4. Producer - consumer count.
5. Storage.

### CODE REVIEW

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ AREA:val1#babdb6 \
+ LINE2:val1#888a85:"Connection Count " \
+ GPRINT:val1:AVERAGE:"Avg \\: %10.2lf" \
+ GPRINT:val1:MIN:"Min \\: %10.2lf" \
+ GPRINT:val1:MAX:"Max \\: %10.2lf\\n"
+
+#
+# Message Count
+#
+report.activemq.message.count.name=ActiveMQ Message Count
```
Comment (j-white) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: message count vs message size (the same goes for the other labels on this graph)

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+#
+# Prducer-Consumer Count
+#
+report.activemq.consumer.producer.count.name=ActiveMQ Producer-Consumer Count
+report.activemq.consumer.producer.count.columns=TtlProdCnt,TtlConsumerCnt
+report.activemq.consumer.producer.count.type=interfaceSnmp
+report.activemq.consumer.producer.count.command=--title="ActiveMQ Producer-Consumer Count" \
+ --units-exponent=0 \
+ --vertical-label="Message Count" \
```
Comment (j-white) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: produce-consumer count?

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ COMMENT:"Producer Consumer Count\\n" \
+ LINE2:val1#ad7fa8:"Min " \
+ GPRINT:val1:MIN:"%10.2lf\\n" \
+ LINE2:val2#729fcf:"Max " \
+ GPRINT:val2:MAX:"%10.2lf\\n" \
+
+#
+# Storage
+#
+report.activemq.storage.count.name=ActiveMQ Storage
```
Comment (j-white) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: This graph doesn't look right, putting the limit and the % used on the same graph throws the scale off.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ COMMENT:"Storage Count\\n" \
+ LINE2:val1#ad7fa8:"Min " \
+ GPRINT:val1:MIN:"%10.2lf\\n" \
+ LINE2:val2#729fcf:"Max " \
+ GPRINT:val2:MAX:"%10.2lf\\n" \
+
+#
+# Temp Storage
+#
+report.activemq.temp.storage.count.name=ActiveMQ Temp Storage
```
Comment (j-white) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: This one doesn't look right either.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties (hunk 1 above)
Comment (pk015603) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: Taken care.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties (hunk 2 above)
Comment (pk015603) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: Taken care

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties (hunk 3 above)
Comment (pk015603) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: Separated Limit and percentage in a different graphs.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties (hunk 4 above)
Comment (pk015603) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/activemq-graphs.properties]: Separated Limit and percentage in a different graphs.

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by soleger** [comment]
Closed in favor of rebased PR #1251.

**Comment by opennms-bamboo** [comment]
Welcome Pavan (pavanrp1@gmail.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Pavan (pavanrp1@gmail.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Pull request closed without merging**